moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_robot_bt.h

```diff
@@ -0,0 +1,120 @@
+                           const AllowedCollisionMatrix& acm) const override;
+  void checkOtherCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state1,
+                           const robot_state::RobotState& state2, const CollisionRobot& other_robot,
+                           const robot_state::RobotState& other_state1,
+                           const robot_state::RobotState& other_state2) const override;
+  void checkOtherCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state1,
+                           const robot_state::RobotState& state2, const CollisionRobot& other_robot,
+                           const robot_state::RobotState& other_state1, const robot_state::RobotState& other_state2,
+                           const AllowedCollisionMatrix& acm) const override;
+
+  void distanceSelf(const DistanceRequest& req, DistanceResult& res,
+                    const robot_state::RobotState& state) const override;
+
+  void distanceOther(const DistanceRequest& req, DistanceResult& res, const robot_state::RobotState& state,
+                     const CollisionRobot& other_robot, const robot_state::RobotState& other_state) const override;
+
+protected:
+  /** \brief Updates the poses of the objects in the manager according to given robot state. */
+  void updateTransformsFromState(const robot_state::RobotState& state) const;
+  void updateTransformsFromStateCCD(const robot_state::RobotState& state1, const robot_state::RobotState& state2) const;
+
+  void updatedPaddingOrScaling(const std::vector<std::string>& links) override;
+
+  void addAttachedOjectsToManager(const robot_state::RobotState& state) const;
+
+  void getAttachedBodyObjects(const robot_state::AttachedBody* ab, std::vector<void*>& geoms) const;
+
+  void checkSelfCollisionHelper(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state,
+                                const AllowedCollisionMatrix* acm) const;
+
+  void checkSelfCollisionCCDHelper(const CollisionRequest& req, CollisionResult& res,
+                                   const robot_state::RobotState& state1, const robot_state::RobotState& state2,
+                                   const AllowedCollisionMatrix* acm) const;
+
+  void checkOtherCollisionHelper(const CollisionRequest& req, CollisionResult& res,
+                                 const robot_state::RobotState& state, const CollisionRobot& other_robot,
+                                 const robot_state::RobotState& other_state, const AllowedCollisionMatrix* acm) const;
+
+  /** @brief Bullet collision manager taken from tesseract*/
+  mutable tesseract::tesseract_bullet::BulletDiscreteBVHManager bt_manager_;
+  // mutable tesseract::tesseract_bullet::BulletDiscreteSimpleManager bt_manager_;
+
+  mutable tesseract::tesseract_bullet::BulletCastBVHManager bt_manager_CCD_;
+
+  mutable const AllowedCollisionMatrix* acm_;
```

Comment:
IMO, `mutable` should be reserved for things like `mutex`es and conditional variables, where the state of the object doesn't really change. Don't think `acm_` actually falling into that category though.